### ISSUE TITLE
Discusion: Other storage repositories?

### ISSUE THREAD

**Issue by harrison314** [harrison314/BouncyHsm#2]
Are additional storages desired?

Is LiteDb not enough, or is remote/shared storage required?

Example MySQL storage: [MySqlPersistentRepository.zip](https://github.com/harrison314/BouncyHsm/files/12691055/MySqlPersistentRepository.zip)

**Comment by Q-efx** [comment]
I dont think we need other storage options, its not meant for production, right?

**Comment by harrison314** [comment]
> I dont think we need other storage options, its not meant for production, right?

Yes. BouncyHsm is not intended for production deployments.

The idea of ​​other repositories occurred to me more for the sharing of keys between instances. But I haven't needed it yet while using BouncyHsm (about a year on several test environments).

**Comment by Q-efx** [comment]
Ah for the purpose of sharing keys, I´m not familiar with litedb. Can it do what sqlite can do? Like easy multi write / read? Then mounting via share should be enough?

**Comment by psavva** [comment]
LiteDb supports "Connection type (“direct” or “shared”)
Would a shared connection not be enough for sharing of keys?

Direct: The engine will open the datafile in exclusive mode and will keep it open until Dispose(). The datafile cannot be opened by another process. This is the recommended mode because it’s faster and cachable.

Shared: The engine will be close the datafile after each operation. Locks are made using Mutex. This is more expensive but you can open same file from multiple processes.

**Comment by harrison314** [comment]
@psavva thanks for the tip with LiteDb. But in the context of BouncyHSM, sharing the LiteDb file is not so important, because multiple PKCS11 libraries can connect (over TCP) to BouncyHSM even from remote nodes.

**Comment by harrison314** [comment]
In the two years of Bouncy HSM's existence, I have not seen a request for key sharing via storage. So I am closing this issue.
